viewer/packages/360-images/src/icons/IconCollection.ts

```diff
@@ -32,7 +31,7 @@ export class IconCollection {
     const sharedTexture = this.createOuterRingsTexture();
     const iconSpriteRadius = 0.5;
     const iconsSprites = new InstancedIconSprite(
-      points.length,
+      points.length * 2,
```

Comment:
I'll create a variable with a descriptive name. But this is essentially an upper bound of how many points could be visualized at the same time, which is every icon, and every node in the octree which is numIcons + numIcons / 2 + numIcons / 4 =~ numIcons * 2. This is of course extremly unlikely (and probably even technically impossible), but it prevents us ever accessing out-of-bounds in the underlying buffer.

Reply:
When I think about it I believe we actually are technically bounded by just points.length. Since we use an actual icon to represent the cluster and we filter out duplicates. The intuition being, that every cluster is represented by some pre-existing icon and that no icons are visualized twice.